pkg/controller/controller.go

```diff
@@ -533,11 +551,34 @@ func (ctrl *resizeController) patchClaim(oldPVC, newPVC *v1.PersistentVolumeClai
 	return updatedClaim, nil
 }
 
-func (ctrl *resizeController) updatePVCapacity(pv *v1.PersistentVolume, newCapacity resource.Quantity) error {
+func (ctrl *resizeController) deletePreResizeCapAnnotation(pv *v1.PersistentVolume) error {
+	// if the pv does not have a resize annotation skip the entire process
+	if !metav1.HasAnnotation(pv.ObjectMeta, util.AnnPreResizeCapacity) {
+		return nil
+	}
+	pvClone := pv.DeepCopy()
+	delete(pvClone.ObjectMeta.Annotations, util.AnnPreResizeCapacity)
+
+	_, err := ctrl.patchPersistentVolume(pv, pvClone)
+	return err
+}
+
+func (ctrl *resizeController) updatePVCapacity(pv *v1.PersistentVolume, oldCapacity, newCapacity resource.Quantity, fsResizeRequired bool) error {
 	klog.V(4).Infof("Resize volume succeeded for volume %q, start to update PV's capacity", pv.Name)
 	newPV := pv.DeepCopy()
 	newPV.Spec.Capacity[v1.ResourceStorage] = newCapacity
 
+	if fsResizeRequired {
+		// if the pv already has a resize annotation skip
+		if metav1.HasAnnotation(pv.ObjectMeta, util.AnnPreResizeCapacity) {
```

Comment:
I think this is problematic - if I expand a PV from 10 to 20GB and while expansion is pending on the node, I expand it again to 40GB then second expansion won't be recorded in the pv spec?

Reply:
You are definitely right on this. The function should not return before updating `pv.Spec.Capacity`